test/e2e_node/podresources_test.go

```diff
@@ -0,0 +1,380 @@
+func (tpd *testPodData) deletePod(f *framework.Framework, podName string) {
+	_, ok := tpd.PodMap[podName]
+	if !ok {
+		return
+	}
+	deletePodSyncByName(f, podName)
+	delete(tpd.PodMap, podName)
+}
+
+func expectPodResources(cli kubeletpodresourcesv1.PodResourcesListerClient, expectedPodsWithResources, expectedPodsWithoutResources int) {
+	gomega.EventuallyWithOffset(1, func() error {
+		podResources, noResources := getPodResources(cli)
+		if len(podResources) != expectedPodsWithResources {
+			return fmt.Errorf("pod with resources: expected %d found %d", expectedPodsWithResources, len(podResources))
+		}
+		if len(noResources) != expectedPodsWithoutResources {
+			return fmt.Errorf("pod WITHOUT resources: expected %d found %d", expectedPodsWithoutResources, len(noResources))
+		}
+		return nil
+	}, time.Minute, 10*time.Second).Should(gomega.BeNil())
+}
+
+func podresourcesListTests(f *framework.Framework, cli kubeletpodresourcesv1.PodResourcesListerClient, sd *sriovData) {
+	var podResources []*kubeletpodresourcesv1.PodResources
+	var noResources []*kubeletpodresourcesv1.PodResources
+	var tpd *testPodData
+
+	ginkgo.By("checking the output when no pods are present")
+	expectPodResources(cli, 0, 1) // sriovdp
+
+	tpd = newTestPodData()
+	ginkgo.By("checking the output when only pods which don't require resources are present")
+	tpd.createPodsForTest(f, []podDesc{
+		{
+			podName: "pod-00",
+		},
+		{
+			podName: "pod-01",
+		},
+	})
+	expectPodResources(cli, 0, 2+1) // test pods + sriovdp
+	tpd.deletePodsForTest(f)
+
+	tpd = newTestPodData()
+	ginkgo.By("checking the output when only a subset of pods require resources")
+	tpd.createPodsForTest(f, []podDesc{
+		{
+			podName: "pod-00",
+		},
+		{
+			podName:        "pod-01",
+			resourceName:   sd.resourceName,
+			resourceAmount: "1",
+		},
+		{
+			podName: "pod-02",
+		},
+		{
+			podName:        "pod-03",
+			resourceName:   sd.resourceName,
+			resourceAmount: "1",
+		},
+	})
+	expectPodResources(cli, 2, 2+1) // test pods + sriovdp
+	// TODO check for specific pods
+	tpd.deletePodsForTest(f)
+
+	tpd = newTestPodData()
+	ginkgo.By("checking the output when creating pods which require resources between calls")
+	tpd.createPodsForTest(f, []podDesc{
+		{
+			podName: "pod-00",
+		},
+		{
+			podName:        "pod-01",
+			resourceName:   sd.resourceName,
+			resourceAmount: "1",
+		},
+		{
+			podName: "pod-02",
+		},
+	})
+	podResources, noResources = getPodResources(cli)
+	framework.ExpectEqual(len(podResources), 1)
+	framework.ExpectEqual(len(noResources), 2+1) // test pods + sriovdp
+	// TODO check for specific pods
+
+	tpd.createPodsForTest(f, []podDesc{
+		{
+			podName:        "pod-03",
+			resourceName:   sd.resourceName,
+			resourceAmount: "1",
+		},
+	})
+	podResources, noResources = getPodResources(cli)
+	framework.ExpectEqual(len(podResources), 2)
+	framework.ExpectEqual(len(noResources), 2+1) // test pods + sriovdp
+	// TODO check for specific pods
```

Comment:
am I missing something?  there is nothing that appears to verify that cpus are actually appearing in the response?

Reply:
right, missing (lost in rebases). Will take care.